### PR TITLE
Sof 1531/Able to activate rundown with keyboard

### PR DESCRIPTION
**Pull request by RasmusAlbrektsen**
This PR introduces the temporary possibility of using the usual key to activate a rundown from the Rundown View. It also includes the addition of activation dialogs upon such an action.

### CODE REVIEW

File: src/app/rundown/components/rundown/rundown.component.ts [hunk 1]
```diff
@@ -13,14 +14,23 @@ import { Segment } from '../../../core/models/segment'
 })
 export class RundownComponent implements OnInit, OnDestroy {
 
+  //TODO: Remove this temporary implementation once keyboard shortcuts are added
+  @HostListener('document:keypress', ['$event'])
+  handleKeyboardEvent2(event: KeyboardEvent) {
```
Comment (LindvedKrvang) [inline src/app/rundown/components/rundown/rundown.component.ts:19]: Why '2'?

File: src/app/shared/services/dialog.service.ts [hunk 2]
```diff
@@ -33,4 +33,20 @@ export class DialogService {
       onOk();
     });
   }
+
+  public openActivationDialog(nameOfRundownToActivate: string, onOk: () => void): void {
```
Comment (LindvedKrvang) [inline src/app/shared/services/dialog.service.ts:37]: This isn't an optimal way of using the DialogService, but we talked about this already, so just leaving the comment as a reminder

File: src/app/rundown/components/rundown/rundown.component.ts [hunk 3]
```diff
@@ -38,25 +48,20 @@ export class RundownComponent implements OnInit, OnDestroy {
...
     }
-    this.rundownService.activate(this.rundown.id).subscribe()
-  }
-
-  public deactivateRundown(): void {
-    if (!this.rundown?.id) {
-      return
+    if (!this.rundown.isActive) {
+      this.dialogService.openActivationDialog(this.rundown.name, () => this.activateRundown())
     }
```
[the remark this answers is not in the document]
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown/rundown.component.ts]: Changed to fit the suggestion!

File: src/app/rundown/components/rundown/rundown.component.ts (hunk 1 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown/rundown.component.ts]: No idea actually, it has been removed!

File: src/app/shared/services/dialog.service.ts (hunk 2 above)
Comment (RasmusAlbrektsen) [reply src/app/shared/services/dialog.service.ts]: Changed to fit what we talked about!

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts [hunk 4]
```diff
@@ -4,8 +4,9 @@ import { Rundown } from '../../../core/models/rundown'
 import { instance, mock, when } from '@typestrong/ts-mockito'
 import { RouterModule } from '@angular/router'
 import { RundownService } from '../../../core/abstractions/rundown.service'
+import { DialogService } from '../../../shared/services/dialog.service'
 
-describe(RundownHeaderContextMenuComponent.name, () => {
+describe('HeaderContextMenuComponent', () => {
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts:9]: Why is this changed back to a string?

File: src/app/rundown-overview/components/rundown-overview/rundown-overview.component.ts [hunk 5]
```diff
@@ -41,7 +41,9 @@ export class RundownOverviewComponent implements OnInit, OnDestroy {
   }
 
   public openDeletionDialog(basicRundown: BasicRundown): void {
-    this.dialogService.openDeletionDialog('Rundown', basicRundown.name, () => this.deleteRundown(basicRundown.id))
+    this.dialogService.createConfirmDialog('Delete Rundown?', `Are you sure you want to delete the Rundown "${basicRundown.name}"?\n\nPlease note: This action is irreversible!`, 'Delete', () =>
```
Comment (KvelaGorrrrnio) [inline src/app/rundown-overview/components/rundown-overview/rundown-overview.component.ts:44]: A note to all of us: We should also look into translation of text in the UI.

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts [hunk 6]
```diff
@@ -14,11 +15,15 @@ describe(RundownHeaderContextMenuComponent.name, () => {
   })
 })
 
-async function configureTestBed(params: { mockedRundownService?: RundownService } = {}): Promise<RundownHeaderContextMenuComponent> {
+async function configureTestBed(params: { mockedRundownService?: RundownService; mockedDialogService?: DialogService } = {}): Promise<RundownHeaderContextMenuComponent> {
```
Comment (KvelaGorrrrnio) [inline src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts:18]: You can also use `,` for attribute separation in type definitions.

**Review by KvelaGorrrrnio — APPROVED** [review]

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts (hunk 4 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts:9]: Must have been a merge-mishap!

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts (hunk 6 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts:18]: I know, but the linter won't let me :-D
image

File: src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts (hunk 4 above)
Comment (RasmusAlbrektsen) [reply src/app/rundown/components/rundown-header-context-menu/rundown-header-context-menu.component.spec.ts]: I have changed it back again!

File: src/app/rundown-overview/components/rundown-overview/rundown-overview.component.ts (hunk 5 above)
Comment (LindvedKrvang) [reply src/app/rundown-overview/components/rundown-overview/rundown-overview.component.ts:44]: We could look into how Angular handles it https://angular.io/guide/i18n-overview

**Review by LindvedKrvang — APPROVED** [review]

File: src/app/shared/services/dialog.service.ts (hunk 2 above)
Comment (LindvedKrvang) [reply src/app/shared/services/dialog.service.ts]: B-e-a-utiful

File: src/app/rundown-overview/components/rundown-overview/rundown-overview.component.ts (hunk 5 above)
Comment (LindvedKrvang) [reply src/app/rundown-overview/components/rundown-overview/rundown-overview.component.ts:44]: I have created a story for it https://tv2cms.atlassian.net/browse/SOF-1640
